source/adapters/level_zero/queue.cpp

```diff
@@ -1192,10 +1192,20 @@ ur_queue_handle_t_::ur_queue_handle_t_(
     }
     return std::atoi(UrRet) != 0;
   }();
+  static const bool useInterruptBasedEvents = [] {
+    const char *UrRet = std::getenv("UR_L0_USE_INTERRUPT_BASED_EVENTS");
+    if (!UrRet) {
+      return true;
+    }
+    return std::atoi(UrRet) != 0;
+  }();
   this->CounterBasedEventsEnabled =
       UsingImmCmdLists && isInOrderQueue() && Device->useDriverInOrderLists() &&
       useDriverCounterBasedEvents &&
       Device->Platform->ZeDriverEventPoolCountingEventsExtensionFound;
+  this->InterruptBasedEventsEnabled = useInterruptBasedEvents &&
+                                      isLowPowerEvents() && isInOrderQueue() &&
```

Comment:
This should be simplified to:
isLowPowerEvents() && isInOrderQueue() &&  Device->useDriverInOrderLists();

The user directly requests for this feature so we don't need another variable to ignore the user.